crates/accelerate/src/uc_gate.rs

```diff
@@ -93,19 +101,17 @@ fn demultiplex_single_uc(
     let r1 = (IM / 2. * (PI / 2. - phi / 2. - x11.arg())).exp();
     let r2 = (IM / 2. * (PI / 2. - phi / 2. + x11.arg() + PI)).exp();
 
-    let r: Matrix2<Complex64> = Matrix2::from_row_slice(&[r1, C_ZERO, C_ZERO, r2]);
-    let (mut diag, mut u) = compute_2x2_eig(r * x * r);
+    let r = Matrix2::new(r1, C_ZERO, C_ZERO, r2);
+    let (diag, mut u) = compute_2x2_eig(r * x * r);
 
     // If d is not equal to diag(i,-i), then we put it into this "standard" form
     // (see eq. (13) in https://arxiv.org/pdf/quant-ph/0410066.pdf) by interchanging
     // the eigenvalues and eigenvectors
     if (diag[0] + IM).abs() < EPS {
-        diag = [diag[1], diag[0]];
-        u = [[u[(0, 1)], u[(0, 0)]], [u[(1, 1)], u[(1, 0)]]].into();
+        u = Matrix2::new(u[(0, 1)], u[(0, 0)], u[(1, 1)], u[(1, 0)]);
```

Comment:
A small comment about this: the line `u = [[u[(0, 1)], u[(0, 0)]], [u[(1, 1)], u[(1, 0)]]].into();` led to wrong results, I suspect it's the column-major vs. row-major ordering.

Reply:
Yeah that's row vs column ordering. With `.into()` it was probably taking arrays of columns not arrays of rows.